data_processing/jobs/silver/silver-with-publish.yml

```diff
@@ -0,0 +1,548 @@
+                minor=$(version_minor)
+                revision=$(version_revision)
+                echo "Last Tag: NOT Present"
+              else
+                IFS='.' read -ra versionParts <<< "${lastTag#v}"
+                major="${versionParts[0]}"
+                minor="${versionParts[1]}"
+                lastrevision="${versionParts[2]}"
+                revision=$((lastrevision + 1))
+                echo "Last Tag: $lastTag"
+              fi
+              newVersion="${major}.${minor}.${revision}"
+              echo "New Version: $newVersion"
+              echo "##vso[task.setvariable variable=major]$major"
+              echo "##vso[task.setvariable variable=minor]$minor"
+              echo "##vso[task.setvariable variable=revision]$revision"
+              echo "##vso[task.setvariable variable=newVersion]$newVersion"
+            displayName: Determine New Version
+
+          # Create a new tag with the incremented version
+          # - script: |
+          #     git config user.name "BuildService"
+          #     git config user.email "builder@${COMPANY}.com"
+          #     git tag -a ${newVersion} -m "Release created by Azure DevOps"
+          #     git push origin ${newVersion}
+          #   displayName: Create New Tag
+
+          - task: Bash@3
+            displayName: Tag Code
+            inputs:
+              targetType: "inline"
+              script: |
+                commit=$(Build.SourceVersion)
+                tag=$(git tag --contains $commit)
+                if [ -z "$tag" ]; then
+                  echo "Tag does not exist for the commit"
+                  git config user.name "BuildService"
+                  git config user.email "builder@${COMPANY}.com"
+                  echo "Creating tag v${newVersion}..."
+                  git tag -a "v${newVersion}" -m "Release created by Azure DevOps"
+                  git push origin "v${newVersion}"
+                  echo "##vso[task.setvariable variable=ShouldCreateRelease]True"
+                else
+                  echo "Tag '$tag' already exists for the commit.Skipping tag creation"
+                  echo "##vso[task.setvariable variable=ShouldCreateRelease]false"
+                fi
+            env:
+              COMPANY: $(company)
+              newVersion: $(newVersion)
+
+          # #           # Create a GitHub release with these packages
+          - task: GitHubRelease@1
+            displayName: Create GitHub Release
+            inputs:
+              gitHubConnection: $(github_release_service_connection)
+              repositoryName: $(github_org)/$(self_repo)
+              tag: v${newVersion}
+              releaseNotesSource: 'inline'
+              releaseNotesInline: "$(major).$(minor).$(revision)"
+              tagSource: 'gitTag'
+              changeLogCompareToRelease: 'lastFullRelease'
+              changeLogType: 'commitBased'
+            condition: eq(variables['ShouldCreateRelease'], 'true')
```

Comment:
[fmt] Consider adding empty lines at the end of .yml files.

Reply:
@satenderrathee

Reply:
This file is not required, will be removing it.

Reply:
Removed